### PR TITLE
Util pkg refactored into sub pkgs

### PR DESCRIPTION
**Pull request by resouer**
ref #15634

Refactor `crlf` `crypto` `deadlock` `diff` to sub pkg

### CODE REVIEW

**Comment by resouer** [comment]
@k8s-bot e2e test this please. ref # #20910

**Comment by resouer** [comment]
[NOTE] #20910 occurs in this PR

File: pkg/util/detector/deadlock-detector.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package util
+package detector
```
Comment (lavalamp) [inline pkg/util/detector/deadlock-detector.go]: Maybe `threading` is a better name?

**Comment by lavalamp** [comment]
lgtm, but can I get you to change 'detector' to 'threading'?

**Comment by resouer** [comment]
@lavalamp Updated to`threading`

**Comment by resouer** [comment]
@k8s-bot unit test this please, this should not fail.

**Comment by resouer** [comment]
@lavalamp Updated tothreading & fixed rebase

**Comment by lavalamp** [comment]
LGTM thanks!

**Pull request merged**